packages/core/src/modules/proofs/formats/presentation-exchange/PresentationExchangeFormatService.ts

```diff
@@ -0,0 +1,566 @@
+
+    // assertionMethod?: Array<string | VerificationMethod>
+
+    // 1. Get the key for this given proof purpose in this DID document
+    const keyId = didResolutionResult.didDocument[proofPurpose] as string[]
+
+    // get keys from the did document section containing the proof purpose
+
+    // 2. Map the Key Id to a key. Key contains publicKey and keyType attributes
+    const privateKey: Key = keyReferenceToKey(didResolutionResult.didDocument, keyId[0])
+
+    // 3. Use the retrieved key to determine proof type
+    const proofType = keyTypeToProofType(privateKey)
+
+    if (!proofType) {
+      throw new AriesFrameworkError(`Unsupported key type: ${privateKey.keyType}`)
+    }
+
+    const params: PresentationSignOptions = {
+      holder: subject.id,
+      proofOptions: {
+        type: proofType,
+        proofPurpose: IProofPurpose.assertionMethod,
+        challenge: requestPresentation.options?.challenge,
+      },
+      signatureOptions: {
+        verificationMethod: (didResolutionResult.didDocument?.authentication[0]).toString(),
+        keyEncoding: KeyEncoding.Base58,
+        privateKey: privateKey.publicKeyBase58,
+      },
+    }
+
+    this.agentContext = agentContext
+
+    const verifiablePresentation = await pex.verifiablePresentationFromAsync(
+      requestPresentation.presentationDefinition,
+      [options.proofFormats.presentationExchange.formats], // TBD required an IVerifiableCredential[] but AutoSelectCredential returns single credential
+      this.signedProofCallBack.bind(this),
+      params
+    )
+
+    const attachId = options.id ?? uuid()
+
+    const format = new ProofFormatSpec({
+      attachmentId: attachId,
+      format: V2_PRESENTATION_EXCHANGE_PRESENTATION,
+    })
+
+    const attachment = new Attachment({
+      id: attachId,
+      mimeType: 'application/json',
+      data: new AttachmentData({
+        json: JsonTransformer.toJSON(verifiablePresentation),
+      }),
+    })
+
+    return { format, attachment }
+  }
+  public async processPresentation(agentContext: AgentContext, options: ProcessPresentationOptions): Promise<boolean> {
+    if (!options.formatAttachments) {
+      throw Error('Presentation  missing while processing presentation in presentation exchange service.')
+    }
+
+    const requestFormat = options.formatAttachments.request.find(
+      (x) => x.format.format === V2_PRESENTATION_EXCHANGE_PRESENTATION_REQUEST
+    )
+
+    const proofFormat = options.formatAttachments.presentation.find(
+      (x) => x.format.format === V2_PRESENTATION_EXCHANGE_PRESENTATION
+    )
+
+    const proofAttachment = requestFormat?.attachment.getDataAsJson<Attachment>() ?? null
+
+    const requestMessage: RequestPresentationExchangeOptions =
+      proofAttachment as unknown as RequestPresentationExchangeOptions
+
+    const proofPresentationRequestJson = proofFormat?.attachment.getDataAsJson<Attachment>() ?? null
```

Comment:
The return type is not an attachment

```suggestion
    const proofPresentationRequestJson = proofFormat?.attachment.getDataAsJson()
```